onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc

```diff
@@ -1740,14 +1747,22 @@ static const std::unordered_map<std::string_view, const HandlerInfo&> handler_ma
     {"Reshape", reshape_handler},
 };
 
+constexpr bool IsOnnxDomain(std::string_view domain) {
+  return (domain == onnxruntime::kOnnxDomain) || (domain == onnxruntime::kOnnxDomainAlias);
+}
+
+constexpr bool IsMSDomain(std::string_view domain) {
+  return domain == onnxruntime::kMSDomain;
+}
+
 static const HandlerInfo* GetHandler(api::NodeRef& node, const HandlerMap& extended_handlers) {
   std::string key;
   auto domain = node.Domain();
   auto op_type = node.OpType();
 
-  if (domain == onnxruntime::kOnnxDomain || domain == onnxruntime::kOnnxDomainAlias) {
+  if (IsOnnxDomain(domain)) {
     key = std::string(op_type);
-  } else if (domain == onnxruntime::kMSDomain) {
+  } else if (IsMSDomain(domain)) {
```

Comment:
Do we need this test vs. just using the domain either way?

Reply:
Yes, I believe we do. The ONNX domain could either be an empty string or "ai.onnx". In either case, the handler_map key for ONNX ops is expected to be just the op type.

If we instead chose to always use the domain, then we would have to add duplicate entries to the handler map to account for "ai.onnx", which I'm ambivalent about:
```C++
static const std::unordered_map<std::string_view, const HandlerInfo&> handler_map{
    {"Cast", simple_node_handler},
    {"ai.onnx.Cast", simple_node_handler},  // Account for ai.onnx domain

```

Reply:
sorry - I meant the 'else if' test only. can we drop that and the `else` simply uses the domain?

Reply:
Ah I see. We only support onnx and ms_domain ops. Other domains branch to the `else` branch and exit the function.